xml/System.IO/FileStream.xml

```diff
@@ -3727,6 +3727,8 @@ The following example shows how to read from a file asynchronously.
  If the given value is less than the current length of the stream, the stream is truncated. In this scenario, if the current position is greater than the new length, the current position is moved to the last byte of the stream. If the given value is larger than the current length of the stream, the stream is expanded, and the current position remains the same. If the stream is expanded, the contents of the stream between the old and the new length are undefined.  
   
  A stream must support both writing and seeking for `SetLength` to work.  
+
+ This method sets <xref:System.IO.Stream.Position> to `value` and proceeds to set <xref:System.IO.Stream.Length> to `value`, then restores <xref:System.IO.Stream.Position> to its original value. If this method fails, <xref:System.IO.Stream.Position> may not be restored.
```

Comment:
```suggestion
If the operation fails, the current position is undefined.
```
This is describes the current implementation of this method on Windows. The implementation may change in future; or it may be different between OSes. We should avoid describing implementation details in the documentation where possible. The documentation should describe externally observable behavior or invariants instead.